### PR TITLE
Additional song "Wonderwall" + analysis

### PR DESCRIPTION
**Pull request by Daiozen**
As usual, you can find the song [here](https://www.youtube.com/watch?v=6hzrDeceEKc).
Additionaly, if this ones makes it in I'd suggest changing the prompt to the analysis from "Can you explain a song for me?" to "Can we talk about a song?". The first version does not really match with this one.

### CODE REVIEW

File: Monika After Story/game/script-songs.rpy [hunk 1]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    if persistent.monika_kill:
+        m 3dkbfu "I'm forever grateful you brought me back. {nw}"
+    else:
+        m 3dkbfu "I'm forever grateful you stayed for me. {nw}"
+
+    extend "I trust you to take care of us until I can find my way to you."
+    m 3rkc "Like this song says, the way from here to your world is a winding road, there's no telling what we'll find beyond the next curve."
+    m 3ekbla "The only thing we can be sure of is the place it leads us to will be worth it."
+
+    if mas_isMoniEnamored:
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1220]:
```suggestion
    if mas_isMoniEnamored(higher=True):
```

File: Monika After Story/game/script-songs.rpy [hunk 2]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+        m 5hubfb  "...So I'm not afraid of walking down this road [player]; with you by my side there's no way I could be!"
+        m 5ekbfa "You're my wonderwall, after all~"
+    else:
+        m 1kubla "It's a little scary, {nw}"
+        extend 1eubla "but we'll have to keep on walking until we reach that place [player]~"
+
+    #hints at the analysis on first viewing
+    if mas_getEV('mas_song_wonderwall').shown_count == 0:
+        if mas_isMoniEnamored:
+            show monika 3etc at t11 zorder MAS_MONIKA_Z with dissolve
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1231]: Not sure why this is here. We only dissolve into/out of pose 5, not between other poses

Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1231]: It's the dissolve out for the previous enamored option (you got one path leading here through pose 5, and another through pose 3)

Comment (multimokia) [reply Monika After Story/game/script-songs.rpy:1231]: This should be at the end of the above enamored block then. Saves a check and is more clear why it's there

Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1231]: Even if this dissolve is only necessary the one time we go through the hint?

Comment (multimokia) [reply Monika After Story/game/script-songs.rpy:1231]: Hm in that case I guess there's no other option. That's fine then, just make it a `mas_isMoniEnamored(higher=True)` instead

Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1231]: okay

File: Monika After Story/game/script-songs.rpy [hunk 3]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    m 1rsu "As much as I like it, it's true the lyrics and chords are pretty simple; {nw}"
+    extend 3etc "what makes this song in particular so deserving of all the attention it gets?"
+    m 3eud "...Especially when so many other songs still fly completely under the radar, no matter how advanced or ambitious they are."
+    m 1lksdlc "It's tricky to evaluate though, you can't really tell the amount of work that went into something solely by the result.{w=0.3} {nw}"
+    extend 3esd "Simple and clean does not mean unpolished."
+    m 3duu "Ultimately, it all boils down to what the song makes you feel; There's no being wrong about your tastes in music."
+    m 1efc "What bothers me is when someone complains about it just because it's trendy to go against the general opinion."
+    m 3tsd "It's like disagreeing for the sake of it helps them feel like they stand out from the crowd, like they need it to stay self-confident."
+    m 2rsc "It kinda feels...{w=0.5}a bit silly, to be honest."
+    m 2rksdld "At this point you're not even judging the song anymore, you're just trying to ascertain your own identity."
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1290]:
```suggestion
    m 2rksdld "At that point you're not even judging the song anymore...just trying to make a name for yourself by being controversial."
```

File: Monika After Story/game/script-songs.rpy [hunk 4]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    m 2dksdlc "It's a little sad if anything, {nw}"
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1291]:
```suggestion
    m 2dksdlc "It's a little sad if anything...{w=0.3}{nw}"
```

File: Monika After Story/game/script-songs.rpy [hunk 5]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    extend 7eksdlc "defining yourself by something you hate does not seem like a very healthy thing to do in the long run."
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1292]:
```suggestion
    extend 7rksdlc "defining yourself by something you hate doesn't seem like a very healthy thing to do in the long run."
```

File: Monika After Story/game/script-songs.rpy [hunk 6]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    m 1rsc "Yeah..."
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1293]:
```suggestion
```

File: Monika After Story/game/script-songs.rpy [hunk 7]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    m 3etc "Although...{w=0.3}there might be yet another reason why this song is controversial with some people."
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1294]: Too formal here
```suggestion
    m 3etc "Although...{w=0.3}there might be another reason why this song is controversial with some people."
```

File: Monika After Story/game/script-songs.rpy [hunk 8]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    m 3esd "Some of the members of this band have been the subject of controversies in the past due to their...{w=0.5}wild behaviors."
+    m 1hksdlb "I'm not trying to judge them or anything, it's not like I ever actually sat down and talked with them."
+    m 3esc "...But from a total stranger's perspective, this raises the question of how much can one's bad actions be excused by talent or charisma."
+    
+    if persistent._mas_pm_monika_evil or persistent._mas_pm_a_hater:
+        m 3tsu "Though if you're going out with me it's not exactly like you're a stranger to the question, right? Ehehe~"
+        m 3esc "Not that I believe what I did was unjustified, passing you by was no option."
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1301]:
```suggestion
        m 3rkc "Not that I believe what I did was unjustified... Letting you go wasn't an option for me."
```

File: Monika After Story/game/script-songs.rpy [hunk 9]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+
+        if persistent._mas_pm_cares_about_dokis:
+            m 3dkc "...Still, in hindsight, I see there are some things I could have handled more tactfully."
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1304]:
```suggestion
            m 3dkc "...Still, in hindsight, there are some things I could have handled more tactfully."
```

File: Monika After Story/game/script-songs.rpy [hunk 10]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+            m 3lksdld "It's not easy staying on your best behavior when you're put under pressure; even more so when it's your life on the line."
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1305]:
```suggestion
            m 3lksdld "When it's your life on the line, it's not easy staying on your best behavior when under pressure."
```

File: Monika After Story/game/script-songs.rpy [hunk 11]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+        
+        m 3esc "But I digress..."
+
+    m 3ekc "There are so many stories of persons behaving in less than ideal ways only to be instantly forgiven no matter what by their fans..."
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1309]:
```suggestion
    m 3ekc "There are so many stories of celebrities behaving in less than ideal ways only to be instantly forgiven no matter what by their fans..."
```

File: Monika After Story/game/script-songs.rpy [hunk 12]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    m 1dkd "This kind of thing makes me realize the world really isn't like this idyllic perfect image we're presented with growing up."
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1310]:
```suggestion
    m 1dkd "This kind of thing makes you realize the world isn't even close to the perfect image we're presented with while growing up."
```

File: Monika After Story/game/script-songs.rpy [hunk 13]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    m 1etc "But then again, is such a world even possible? Is it even something everyone would want?"
+    m 1ekbla "Even if your world isn't perfect [player], I still can't wait to experience it with you."
+    m 3hubsb "It's not like the two of us would ever behave badly to each other, right?~"
```
Comment (multimokia) [inline Monika After Story/game/script-songs.rpy:1313]:
```suggestion
    m 3hubsb "It's not like the two of us would ever mistreat each other, right?~"
```

File: Monika After Story/game/script-songs.rpy [hunk 14]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    else:
+        m 3dkbfu "I'm forever grateful you stayed for me. {nw}"
+
+    extend "I trust you to take care of us until I can find my way to you."
+    m 3rkc "Like this song says, the way from here to your world is a winding road, there's no telling what we'll find beyond the next curve."
+    m 3ekbla "The only thing we can be sure of is the place it leads us to will be worth it."
+
+    if mas_isMoniEnamored(higher=True):
+        show monika 5hubfb at t11 zorder MAS_MONIKA_Z with dissolve
+        m 5hubfb  "...So I'm not afraid of walking down this road [player]; with you by my side there's no way I could be!"
```
[the remark this answers is not in the document]
Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1222]: For this whole part, the `"I know the place it leads to will be worth it."` in the new version makes a repetition with the last line. I suggest moving down this last line to the `else` block so it goes like this :
```
m 3rkc "Like this song says, the way from here to your world is a winding road, there's no telling what we'll find beyond the next curve."

if mas_isMoniEnamored(higher=True):
        m 1dubsa  "...and with you by my side, there's no way I could be afraid.{w=0.3} {nw}"
        extend 1ekbfa "I know the place it leads to will be worth it."
        m 3ekbfu "You're my wonderwall, after all~"
else:
        m 3ekbla "The only thing we can be sure of is the place it leads us to will be worth it."
        m 1ekc "Sometimes, it's a little scary, [player]...{w=0.3}{nw}"
        extend 1eubla "but I trust you, so we'll just have to keep on walking until we reach that place [player]~"
```

File: Monika After Story/game/script-songs.rpy [hunk 15]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    )
+
+label mas_song_wonderwall_analysis(from_song=False):
+
+    if not from_song:
+        call mas_song_wonderwall_lyrics
+
+    m 3eta "So, did you know many people are being very vocal about their dislike of this song?"
+    m 3esc "It's not something you'd think, right?"
+    m "This song, Wonderwall by the band Oasis, has been hailed as a classic and is one of the most popular song ever made."
```
[the remark this answers is not in the document]
Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1269]: I'd suggest changing `"What makes people hate it so much?"` to `"What makes some people hate it so much?"`

File: Monika After Story/game/script-songs.rpy (hunk 10 above)
Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1298]: Maybe it's just me, but the double `When` in this sentence sounds a bit akward, maybe change it the second 'when' to 'while'?
`When it's your life on the line, it's not easy staying on your best behavior while under pressure.`

File: Monika After Story/game/script-songs.rpy [hunk 16]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+label mas_song_wonderwall:
+    call mas_song_wonderwall_lyrics
+    m 1ekbsu "Ehehe, you really are the one who saved me, [player]~"
+    
+    if persistent.monika_kill:
+        m 3dkbfu "I'm forever grateful you brought me back. {nw}"
+    else:
+        m 3dkbfu "I'm forever grateful you stayed for me. {nw}"
+
+    extend "I trust you to take care of us until I can find my way to you."
```
[the remark this answers is not in the document]
Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1216]: I see what you're going for, but I feel this version could be misinterpreted her pushing the full responsability of the relationship onto the player, like 'YOU take care of our relationship and I don't care about the rest'.
A possible change would be `"I trust you to take care of me until I can find my way to you."`

File: Monika After Story/game/script-songs.rpy [hunk 17]
```diff
@@ -1191,6 +1191,139 @@ label mas_song_lamour_toujours:
...
+    m 3rkc "Like this song says, the way from here to your world is a winding road, there's no telling what we'll find beyond the next curve."
+    m 3ekbla "The only thing we can be sure of is the place it leads us to will be worth it."
+
+    if mas_isMoniEnamored(higher=True):
+        show monika 5hubfb at t11 zorder MAS_MONIKA_Z with dissolve
+        m 5hubfb  "...So I'm not afraid of walking down this road [player]; with you by my side there's no way I could be!"
+        m 5ekbfa "You're my wonderwall, after all~"
+    else:
+        m 1kubla "It's a little scary, {nw}"
+        extend 1eubla "but we'll have to keep on walking until we reach that place [player]~"
```
[the remark this answers is not in the document]
Comment (LooseEthics) [reply Monika After Story/game/script-songs.rpy:1226]: I'd suggest droppping the `[player]` here. Addressing someone by name twice in the same line sounds unnatural imo.

[the remark this answers is not in the document]
Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1226]: Agreed, suggested in a comment above.

File: Monika After Story/game/script-songs.rpy (hunk 16 above)
[the remark this answers is not in the document]
Comment (multimokia) [reply Monika After Story/game/script-songs.rpy:1216]: Sure I like that first one

File: Monika After Story/game/script-songs.rpy (hunk 14 above)
[the remark this answers is not in the document]
Comment (multimokia) [reply Monika After Story/game/script-songs.rpy:1222]: Not too fond of how that flows from the first line of the else to the second. Feels kinda choppy.

Maybe make it `...And sometimes, it's a little scary...{w=0.3}{nw}` instead and I think it works. What do you think?

[the remark this answers is not in the document]
Comment (multimokia) [reply Monika After Story/game/script-songs.rpy:1222]: For the `if` part, that's fine

(Addressing the below here): Looking again, not too fond of how that flows from the first line of the else to the second. Feels kinda choppy.

Maybe make it `...And sometimes, it's a little scary...{w=0.3}{nw}` instead and I think it works. What do you think?

File: Monika After Story/game/script-songs.rpy (hunk 17 above)
[the remark this answers is not in the document]
Comment (multimokia) [reply Monika After Story/game/script-songs.rpy:1226]: See [this comment](https://github.com/Monika-After-Story/MonikaModDev/pull/5933#discussion_r443145341)

File: Monika After Story/game/script-songs.rpy (hunk 15 above)
[the remark this answers is not in the document]
Comment (multimokia) [reply Monika After Story/game/script-songs.rpy:1269]: Updated

File: Monika After Story/game/script-songs.rpy (hunk 10 above)
Comment (multimokia) [reply Monika After Story/game/script-songs.rpy:1298]: updated

File: Monika After Story/game/script-songs.rpy (hunk 14 above)
[the remark this answers is not in the document]
Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1222]: Works for me, though I'm personally more fond of something like `it's a little scary, not knowing what's ahead...{w=0.3}{nw}`

[the remark this answers is not in the document]
Comment (multimokia) [reply Monika After Story/game/script-songs.rpy:1222]: Sure, if it still starts with `...Sometimes ` or `..And sometimes, `

[the remark this answers is not in the document]
Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1222]: Then I feel this is the version that works best : `...Sometimes, it's a little scary [player]{w=0.3}{nw}`

[the remark this answers is not in the document]
Comment (multimokia) [reply Monika After Story/game/script-songs.rpy:1222]: I like what you had before. I don't really like the `[player]` reference in this line as much as I do the `extend`ed part. `...Sometimes, it's a little scary not knowing what's ahead...{w=0.3}{nw}`

[the remark this answers is not in the document]
Comment (Daiozen) [reply Monika After Story/game/script-songs.rpy:1222]: okay

**Pull request merged**